### PR TITLE
Math Libraries Landing pages

### PR DESCRIPTION
**Pull request by MathiasMagnus**

### CODE REVIEW

File: docs/reference/gpu_libraries/c++_primitives.md [hunk 1]
```diff
@@ -1 +1,40 @@
...
+
+:::::{grid} 1 1 3 3
+:gutter: 1
+
+:::{grid-item-card} rocPRIM
+rocPRIM is an AMD GPU optimized template library of algorithm primitives, like
+transforms, reductions, scans, etc. It also serves as a common back-end for
+similar libraries found inside ROCm.
+
+- [Changelog](https://github.com/ROCmSoftwarePlatform/rocPRIM/blob/develop/CHANGELOG.md)
```
Comment (saadrahim) [inline docs/reference/gpu_libraries/c++_primitives.md:13]: What should the order of these be? I know the rocBLAS example had changelog first. Is it best to have the API reference manual first?

**Review by saadrahim — CHANGES_REQUESTED** [review]
Fix rocBLAS card order as well.

**Review by saadrahim — DISMISSED** [review]

**Comment by MathiasMagnus** [comment]
Rebased.

File: docs/reference/gpu_libraries/blas.md [hunk 2]
```diff
@@ -10,18 +10,40 @@ hipBLAS is a compatiblity layer for GPU accelerated BLAS optimized for AMD GPUs
...
+
+:::{grid-item-card} Tensile
+Tensile is a tool for creating benchmark-driven backend libraries for GEMMs,
+GEMM-like problems and general N-dimensional tensor contractions on a GPU.
+The Tensile library is mainly used as backend library to rocBLAS. Tensile acts as the
+performance backbone for a wide variety of 'compute' applications running on AMD GPUs.
+
 - [API Reference Manual](https://rocmdocs.amd.com/projects/rocBLAS/en/rtd/)
+- [Changelog](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/CHANGELOG.md)
 - [Examples](https://github.com/amd/rocm-examples/tree/develop/Libraries/rocBLAS)
```
Comment (saadrahim) [inline docs/reference/gpu_libraries/blas.md:35]: Don't include rocBLAS examples for Tensile.

File: docs/reference/gpu_libraries/fft.md [hunk 3]
```diff
@@ -5,22 +5,22 @@ ROCm libraries for FFT are as follows:
...
-using rocFFT. hipFFT allows for a common interface for other non AMD GPU
-FFT libraries.
+:::{grid-item-card} rocFFT
+rocFFT is an AMD GPU optimized library for FFT.
 
-- [Changelog](https://github.com/ROCmSoftwarePlatform/hipFFT/blob/develop/CHANGELOG.md)
-- [API Reference Manual](https://rocmdocs.amd.com/projects/hipFFT/en/rtd/)
+- [API Reference Manual](https://rocmdocs.amd.com/projects/rocFFT/en/rtd/)
+- [Changelog](https://github.com/ROCmSoftwarePlatform/rocFFT/blob/develop/CHANGELOG.md)
+- [Examples](https://github.com/amd/rocm-examples/tree/develop/Libraries/rocFFT)
```
Comment (saadrahim) [inline docs/reference/gpu_libraries/fft.md:13]: This doesn't exist yet, please remove.

File: docs/reference/gpu_libraries/math.md [hunk 4]
```diff
@@ -1 +1,13 @@
...
+
+## rocLIB vs. hipLIB
+
+Several libraries are prefixed with either "roc" or "hip".
+The rocLIB variants (such as rocRAND, rocBLAS) are tested and optimized for AMD hardware using supported toolchains.
+The hipLIB variants (such as hipRAND, hipBLAS) are compatibility layers that provide an interface akin to their
+cuLIB (such as cuRAND, cuBLAS) variants while performing static dispatching of API calls to the appropriate
+vendor libraries as their back-ends. Due to their static dispatch nature, support for either vendor is decided
+at compile-time of the hipLIB in question. For dynamic dispatch between vendor implementations, refer to the
+[Orochi](https://github.com/GPUOpen-LibrariesAndSDKs/Orochi) library.
```
Comment (saadrahim) [inline docs/reference/gpu_libraries/math.md:13]: Intro to the other math lib pages, probably using grid cards.

**Review by saadrahim — APPROVED** [review]
Approved with comments. Please resolve them and merge. 

Or I may merge this late today and comments to PR should be addressed in a later change.

**Pull request merged**